applications/ChimeraApplication/custom_utilities/hole_cutting_utility.cpp

```diff
@@ -0,0 +1,438 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Authors:        Aditya Ghantasala, https://github.com/adityaghantasala
+// 					Navaneeth K Narayanan
+//					Rishith Ellath Meethal
+//
+
+#include "hole_cutting_utility.h"
+
+namespace Kratos {
+
+template <int TDim>
+void ChimeraHoleCuttingUtility::CreateHoleAfterDistance(ModelPart& rModelPart,
+                                                        ModelPart& rHoleModelPart,
+                                                        ModelPart& rHoleBoundaryModelPart,
+                                                        const double Distance)
+{
+    KRATOS_TRY;
+    ChimeraHoleCuttingUtility::RemoveOutOfDomainElements<TDim>(
+        rModelPart, rHoleModelPart, ChimeraHoleCuttingUtility::Domain::MAIN_BACKGROUND,
+        Distance, ChimeraHoleCuttingUtility::SideToExtract::INSIDE);
+    ChimeraHoleCuttingUtility::ExtractBoundaryMesh<TDim>(rHoleModelPart, rHoleBoundaryModelPart);
+    KRATOS_CATCH("");
+}
+
+template <int TDim>
+void ChimeraHoleCuttingUtility::RemoveOutOfDomainElements(
+    ModelPart& rModelPart,
+    ModelPart& rRemovedModelPart,
+    const ChimeraHoleCuttingUtility::Domain DomainType,
+    const double OverLapDistance,
+    const ChimeraHoleCuttingUtility::SideToExtract Side)
+{
+    KRATOS_TRY;
+    std::vector<IndexType> vector_of_node_ids;
+    vector_of_node_ids.reserve(rModelPart.NumberOfNodes());
+    std::vector<IndexType> vector_of_elem_ids;
+    vector_of_elem_ids.reserve(rModelPart.NumberOfElements());
+
+    for (auto& i_element : rModelPart.Elements()) {
+        const auto& p_elem = rModelPart.pGetElement(i_element.Id());
```

Comment:
ouch ... This is a `find`!
if you need the pointer then better loop with iterators

Reply:
good point ! changing !